var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -197,7 +197,8 @@ class IntelOneapiMkl(IntelOneApiLibraryPackage):
     provides("fftw-api@3")
     provides("scalapack", when="+cluster")
     provides("mkl")
-    provides("lapack", "blas")
```

Comment:
For MKL, lapack and blas are in the same library so there isn't a way to provide one without the other. If there are multiple providers, then the order in the link line will determine which one is selected. It is possible spack tries to be smart about link order to satisfy conflicting providers, but I am not aware of this capability. @haampie : Do you know?

Looking at this more, I think fftw, mkl, lapack, and blas should all be together.